slick/src/main/scala/slick/util/InternalArrayQueue.scala

```diff
@@ -0,0 +1,175 @@
+package slick.util
+
+import java.util
+
+class InternalArrayQueue[E >: Null <: AnyRef](capacity: Int) {
```

Comment:
It looks like some pieces of it were extracted from old `ManagedArrayBlockingQueue`.